### PR TITLE
feat: largest_power

### PR DESCRIPTION
**Pull request by uday6670**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1406"></a>

### CODE REVIEW

File: math/largest_power.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() 
+{ 
+    test(); // execute the tests
+	return 0; 
+} 
+
```
Comment (Panquesito7) [inline math/largest_power.cpp:78]:
```suggestion
```

File: math/largest_power.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() 
+{ 
+    test(); // execute the tests
+	return 0;
```
Comment (Panquesito7) [inline math/largest_power.cpp:76]:
```suggestion
    test(); // execute the tests
    return 0;
```

File: math/largest_power.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ * @brief Mathematical algorithms
+ */
+namespace math {
+
+	/** 
+	 * Function to calculate largest power
+	 * @param n number
+     * @param p prime number
+	 * @return largest power
+	 */
```
Comment (Panquesito7) [inline math/largest_power.cpp:24]:
```suggestion
	/** 
	 * @brief Function to calculate largest power
	 * @param n number
         * @param p prime number
	 * @returns largest power
	 */ 
```

File: math/largest_power.cpp [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+ */
+namespace math {
+
+	/** 
+	 * Function to calculate largest power
+	 * @param n number
+     * @param p prime number
+	 * @return largest power
+	 */ 
+    int largestPower(int n, int p)
```
Comment (Panquesito7) [inline math/largest_power.cpp:25]:
```suggestion
    uint64_t largestPower(uint32_t n, const uint16_t& p)  
```

File: math/largest_power.cpp [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            n /= p;  
+            x += n;  
+        }  
+        return x;  
+    }
+
+}  // namespace math
+
+/**
+ * Function for testing largestPower function.
```
Comment (Panquesito7) [inline math/largest_power.cpp:42]:
```suggestion
 * @brief Function for testing largestPower function.
```

File: math/largest_power.cpp [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+ * test cases and assert statement.
+ * @returns `void`
+*/
+static void test()
+{
+	int test_case_1 = math::largestPower(5,2);
```
Comment (Panquesito7) [inline math/largest_power.cpp:48]:
```suggestion
	uint8_t test_case_1 = math::largestPower(5,2);
```

File: math/largest_power.cpp [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+	assert(test_case_1==3);
+	std::cout<<"Test 1 Passed!"<<std::endl;
+	
+	int test_case_2 = math::largestPower(10,3);
```
Comment (Panquesito7) [inline math/largest_power.cpp:52]:
```suggestion
	uint16_t test_case_2 = math::largestPower(10,3);
```

File: math/largest_power.cpp [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+	assert(test_case_2==4);
+	std::cout<<"Test 2 Passed!"<<std::endl;
+	
+	int test_case_3 = math::largestPower(25,5);
```
Comment (Panquesito7) [inline math/largest_power.cpp:56]:
```suggestion
	uint32_t test_case_3 = math::largestPower(25,5);
```

File: math/largest_power.cpp [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+	assert(test_case_3==6);
+	std::cout<<"Test 3 Passed!"<<std::endl;
+	
+	int test_case_4 = math::largestPower(27,2);
```
Comment (Panquesito7) [inline math/largest_power.cpp:60]:
```suggestion
	uint32_t test_case_4 = math::largestPower(27,2);
```

File: math/largest_power.cpp [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+	assert(test_case_4==23);
+	std::cout<<"Test 4 Passed!"<<std::endl;
+	
+	int test_case_5 = math::largestPower(7,3);
```
Comment (Panquesito7) [inline math/largest_power.cpp:64]:
```suggestion
	uint16_t test_case_5 = math::largestPower(7,3);
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
👍 Well done!

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/uday6670/C-Plus-Plus/actions

**Comment by uday6670** [comment]
@Panquesito7 and @kvedala  Please review

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/uday6670/C-Plus-Plus/actions

**Comment by uday6670** [comment]
I am enabled Github action.

**Comment by uday6670** [comment]
Please review @ayaankhan98

**Review by Panquesito7 — COMMENTED** [review]
Good work. Now, to trigger the GitHub Actions, make a small commit (e.g.: add a new line after line 9).
If GitHub Actions is enabled, we'll see a commit by GitHub Actions updating `DIRECTORY.md`. 🙂

**Review by Panquesito7 — APPROVED** [review]
Great work. Thank you for your contribution! 🎉 👍

**Review by kvedala — APPROVED** [review]

**Pull request merged**